File: isanyuseractive.go
package main

import (
	"os/exec"
	"strings"
)

func isAnyUserActive() bool {
	cmdOutput, err := exec.Command("loginctl", "list-sessions", "--no-legend").Output()
	if err != nil {
		return true
	}

	return strings.TrimSpace((string(cmdOutput))) != ""
}

File: src/isanyuseractive.rs
use std::{io, process::Command};

pub fn is_any_user_active() -> Result<bool, io::Error> {
    let loginctl_output = Command::new("loginctl")
        .arg("list-sessions")
        .arg("--no-legend")
        .output()?;

    Ok(loginctl_output.status.success() && !String::from_utf8_lossy(&loginctl_output.stdout).trim().is_empty())
}
